Missing `IsString` instance with debug flags enabled

Compiling Agda `2.6.4.1` on GHC `9.4.5` with the cabal flags `debug`, `debug-parsing`, and `debug-serialisation` enabled leads to the following erros:

```
src/full/Agda/Main.hs:291:5: error:
    • No instance for (Data.String.IsString (t0 -> [a0] -> [[Char]]))
        arising from the literal ‘"debug-parsing: enable printing grammars for operator parsing via '-v scope.grammar:10'"’
        (maybe you haven't applied a function to enough arguments?)
    • In the second argument of ‘(:)’, namely
        ‘"debug-parsing: enable printing grammars for operator parsing via '-v scope.grammar:10'"
           "debug-serialisation: extra debug info during serialisation into '.agdai' files"
           []’
      In the expression:
        "debug: enable debug printing ('-v' verbosity flags)"
          : "debug-parsing: enable printing grammars for operator parsing via '-v scope.grammar:10'"
              "debug-serialisation: extra debug info during serialisation into '.agdai' files"
              []
      In an equation for ‘flags’:
          flags
            = "debug: enable debug printing ('-v' verbosity flags)"
                : "debug-parsing: enable printing grammars for operator parsing via '-v scope.grammar:10'"
                    "debug-serialisation: extra debug info during serialisation into '.agdai' files"
                    []
    |
291 |     "debug-parsing: enable printing grammars for operator parsing via '-v scope.grammar:10'"
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/full/Agda/Main.hs:294:5: error:
    • Ambiguous type variable ‘t0’ arising from the literal ‘"debug-serialisation: extra debug info during serialisation into '.agdai' files"’
      prevents the constraint ‘(Data.String.IsString
                                  t0)’ from being solved.
      Probable fix: use a type annotation to specify what ‘t0’ should be.
      Potentially matching instances:
        instance (Data.String.IsString a, MonadIO m) =>
                 Data.String.IsString (TCMT m a)
          -- Defined at src/full/Agda/TypeChecking/Monad/Base.hs:5350:10
        instance (a ~ Char) => Data.String.IsString [a]
          -- Defined in ‘Data.String’
        ...plus 33 instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the first argument of ‘"debug-parsing: enable printing grammars for operator parsing via '-v scope.grammar:10'"’, namely
        ‘"debug-serialisation: extra debug info during serialisation into '.agdai' files"’
      In the second argument of ‘(:)’, namely
        ‘"debug-parsing: enable printing grammars for operator parsing via '-v scope.grammar:10'"
           "debug-serialisation: extra debug info during serialisation into '.agdai' files"
           []’
      In the expression:
        "debug: enable debug printing ('-v' verbosity flags)"
          : "debug-parsing: enable printing grammars for operator parsing via '-v scope.grammar:10'"
              "debug-serialisation: extra debug info during serialisation into '.agdai' files"
              []
    |
294 |     "debug-serialisation: extra debug info during serialisation into '.agdai' files"
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/.github/workflows/cabal.yml
+++ b/.github/workflows/cabal.yml
@@ -128,8 +128,8 @@ jobs:
           description: Linux debug
           ghc-ver: 9.8.1
           os: ubuntu-22.04
-        - cabal-flags: |
-            --enable-tests -f enable-cluster-counting -f debug -c containers>=0.7 --allow-newer=containers
+        - cabal-flags: --enable-tests -f enable-cluster-counting -f debug -f debug-serialisation
+            -f debug-parsing -c containers>=0.7 --allow-newer=containers
           description: Linux containers 0.7
           ghc-ver: 9.8.1
           os: ubuntu-22.04

--- a/Agda.cabal
+++ b/Agda.cabal
@@ -393,7 +393,7 @@ library
   build-depends:
     -- Please keep in alphabetical order!
     , aeson                >= 1.1.2.0   && < 2.3
-    , ansi-terminal        >= 0.9       && < 1.1
+    , ansi-terminal        >= 0.9       && < 1.2
     , array                >= 0.5.2.0   && < 0.6
     , async                >= 2.2       && < 2.3
     , base                 >= 4.12.0.0  && < 4.20

--- a/src/github/workflows/cabal.yml
+++ b/src/github/workflows/cabal.yml
@@ -79,8 +79,9 @@ jobs:
             ## Andreas, 2023-09-28: Test containers-0.7 here which has breaking changes.
             ## Note: -c 'containers >= 0.7' with single quotes does not get communicated properly.
             ## (The single quotes stay, and "-c 'containers" is an option parse error for cabal.)
-            cabal-flags: |
-              --enable-tests -f enable-cluster-counting -f debug -c containers>=0.7 --allow-newer=containers
+            cabal-flags: >-
+              --enable-tests -f enable-cluster-counting -f debug -f debug-serialisation -f debug-parsing
+              -c containers>=0.7 --allow-newer=containers
 
           # macOS with default flags
           - os: macos-12

